Sources/Workspace/DefaultPluginScriptRunner.swift

```diff
@@ -506,22 +508,10 @@ public struct DefaultPluginScriptRunner: PluginScriptRunner {
         }
 #endif
     }
-    
-    /// Cancels all currently running plugins, resulting in an error code indicating that they were interrupted. This is intended for use when the host process is interrupted.
-    public static func cancelAllRunningPlugins() {
-#if !os(iOS) && !os(watchOS) && !os(tvOS)
-        currentlyRunningPlugins.lock.withLock {
-            currentlyRunningPlugins.processes.forEach{
-                $0.terminate()
-            }
-            currentlyRunningPlugins.processes = []
-        }
-#endif
+
+    public func cancel(deadline: DispatchTime) throws {
+        try self.cancellator.cancel(deadline: deadline)
```

Comment:
If I read the underlying code correctly, this will block until the cancellation has happened (or deadline passes), right, so should always be called from a background thread in an interactive environment?

Reply:
yes this call is blocking. we can add a non-blocking variant if useful